shell/common/shell_unittests.cc

```diff
@@ -1129,5 +1132,56 @@ TEST_F(ShellTest, CanDecompressImageFromAsset) {
   DestroyShell(std::move(shell));
 }
 
+TEST_F(ShellTest, OnServiceProtocolGetSkSLsWorks) {
+  // Create 2 dummpy SkSL cache file IE (base32 encoding of A), II (base32
+  // encoding of B) with content x and y.
+  fml::ScopedTemporaryDirectory temp_dir;
+  PersistentCache::SetCacheDirectoryPath(temp_dir.path());
+  std::vector<std::string> components = {"flutter_engine",
+                                         GetFlutterEngineVersion(), "skia",
+                                         GetSkiaVersion(), "sksl"};
+  auto sksl_dir = fml::CreateDirectory(temp_dir.fd(), components,
+                                       fml::FilePermission::kReadWrite);
+  const std::string x = "x";
+  const std::string y = "y";
+  auto x_data = std::make_unique<fml::DataMapping>(
+      std::vector<uint8_t>{x.begin(), x.end()});
+  auto y_data = std::make_unique<fml::DataMapping>(
+      std::vector<uint8_t>{y.begin(), y.end()});
+  ASSERT_TRUE(fml::WriteAtomically(sksl_dir, "IE", *x_data));
+  ASSERT_TRUE(fml::WriteAtomically(sksl_dir, "II", *y_data));
+
+  Settings settings = CreateSettingsForFixture();
+  std::unique_ptr<Shell> shell = CreateShell(settings);
+  ServiceProtocol::Handler::ServiceProtocolMap empty_params;
+  rapidjson::Document document;
+  OnServiceProtocolGetSkSLs(shell.get(), empty_params, document);
+  rapidjson::StringBuffer buffer;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  document.Accept(writer);
+  DestroyShell(std::move(shell));
+
+  // Base64 encoding of x, y are eQ, eA.
+  ASSERT_STREQ(
+      buffer.GetString(),
+      "{\"type\":\"GetSkSLs\",\"SkSLs\":{\"II\":\"eQ==\",\"IE\":\"eA==\"}}");
+
+  // Cleanup files
+  fml::FileVisitor recursive_cleanup = [&recursive_cleanup](
```

Comment:
In a later patch, we should consider putting this in //flutter/testing or create a fixture that sets up an isolate temp directory per test run that gets automatically cleaned up.